superset-frontend/src/dashboard/components/DashboardBuilder/DashboardContainer.tsx

```diff
@@ -58,6 +54,9 @@ const DashboardContainer: FC<DashboardContainerProps> = ({
     ? topLevelTabs.children
     : [DASHBOARD_GRID_ID];
 
+  const min = Math.min(tabIndex, childIds.length - 1);
+  const activeKey = min === 0 ? DASHBOARD_GRID_ID : min.toString();
```

Comment:
Ensures that the key of the first element is always `DASHBOARD_GRID_ID` to avoid expensive mounts/unmounts of the entire dashboard.